src/main/frontend/db/query_dsl.cljs

```diff
@@ -459,7 +466,8 @@ Some bindings in this fn:
                                             (keyword (name x))
                                             x)))
                                    (string/join " ")
-                                   (util/format "(between %s)")))))))
+                                   (util/format "(between %s)"))))
+            (string/replace "#" "#tag "))))
```

Comment:
Bump

Reply:
I just pushed 4eaa7fb82645e84ac8069fe1ef5f93ef4dea8483, which will replace ` #` with ` #tag` and `^#` with `# tag ` and will ignore ` #` in any page references.

Reply:
Another bug - `"for #clojure"` -> `"for #tag clojure"`. I think `string/replace` is going to continue to have edge cases since it lacks context. Could we do this translation in `build-query` and wrap the `page-ref?` calls with a `tag?`?

Reply:
For full-text searches such as `"\"for #clojure\""`, `pre-transform` will not replace the tags. Users need to quote the search like `{{query "for #clojure"}}`, `{{query for #clojure}}` will not work. 

> Could we do this translation in build-query and wrap the page-ref? calls with a tag??

build-query will be called after `read-string`, which doesn't work for our tags, for example, `(reader/read-string "#tag")` returns the error `EOF while reading`.

Reply:
Looks like this is an issue as it can't find the 3 blocks I have with that exact string:
<img width="235" alt="Screen Shot 2023-03-15 at 9 41 20 AM" src="https://user-images.githubusercontent.com/97210743/225326550-8422946a-c145-4ad1-ac4c-719825f59d02.png">
Builder built the query for this as `{{query (and "for #clojure")}}`

Reply:
Should be fixed in [e62e8fb](https://github.com/logseq/logseq/pull/8774/commits/e62e8fb844caf40939e223f6c3f56fc90592aba3).

Reply:
Works great! 🚢